go/vt/tabletserver/tx_executor.go

```diff
@@ -68,13 +74,16 @@ func (txe *TxExecutor) CommitPrepared(dtid string) error {
 	if conn == nil {
 		return nil
 	}
-	defer txe.qe.txPool.LocalConclude(txe.ctx, conn)
-	err := txe.qe.twoPC.DeleteRedo(txe.ctx, conn, dtid)
+	// We have to use a context that will never give up,
+	// even if the original context expires.
+	ctx := context.Background()
```

Comment:
There are ways to copy the span info of the original context to this one, this could be valuable here. I believe the code would be:
ctx := trace.CopySpan(context.Background(), txe.ctx)